src/app/app.component.ts

```diff
@@ -1,9 +1,22 @@
 import { Component } from '@angular/core';
 import 'rxjs/add/operator/filter';
+import { MatIconRegistry } from '@angular/material/icon';
+import { DomSanitizer } from '@angular/platform-browser';
 
 @Component({
   selector: 'app-root',
   templateUrl: './app.component.html',
   styleUrls: ['./app.component.scss'],
 })
-export class AppComponent {}
+export class AppComponent {
+  constructor(private matIconRegistry: MatIconRegistry, private domSanitizer: DomSanitizer) {
```

Comment:
Hi dude,

Why this code in the constructor? Aren't the SVG's behave as regular images and can be set in the assets folder?

Reply:
The SVGs are not standard graphical format as they include code.. Yes, they are located in the assets folder.
the main issues is that in order to use them in Angular Materials, you need to load them in advance. Product asked to override the default icons for the male/female in the dialog of the student details. This is the way to do it.. if you want to move this code elsewhere, NP. let me know where and I'll move.

Thanks, Kfir

Reply:
I think that we need to put a function in the OnInit and to initialise these icons there. What do you think?

Reply:
Done. I tried to put it on the OnInit of the StudentDetails.. page and it didn;t work. So I klept it on the AppComponent but removed it from the Constructor into the ngOnInit.